astropy/cosmology/tests/test_core.py

```diff
@@ -219,11 +219,13 @@ def test_name(self, cosmo):
         assert cosmo.name == self.cls_kwargs["name"]  # test has expected value
 
         # immutable
-        match = (
-            "can't set"
-            if PYTHON_LT_3_11
-            else f"property 'name' of {cosmo.__class__.__name__!r} object has no setter"
-        )
+        if sys.version_info < (3, 11):
```

Comment:
note to reviewers: the reason I also refactored the ternary out is that, to the best of my knowledge, UP036 auto cleanup only works for if/else patterns. That is, unless the ruff reimplementation was made smarter than my original code.